### PR TITLE
Redux overhaul

### PR DESCRIPTION
**Pull request by tspeed90**
Redux manages state now.
Session token is removed from state and is just a property on our app class.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,16 @@
-# mad-skillzzzz
+# Quiz Game (in React!)
```
Comment (haydnba) [inline README.md]: I guess you're right... 😔

File: src/components/app.js [hunk 2]
```diff
@@ -4,67 +4,59 @@ import Card from './card';
...
-        this.setState({ quizData: quiz.results })
-      );
+      const { setQuizData, markCategorySelected } = this.props;
+      getQuiz(categoryId, this.sessionToken)
+        .then(quizData => setQuizData(quizData.results))
+        .then(() => markCategorySelected());
     };
   }
 
   restartGame() {
```
[the remark this answers is not in the document]
Comment (haydnba) [reply src/components/app.js]: I'm referring to the restart game function btw - GitHub makes it look like I really care about something else...

File: README.md (hunk 1 above)
Comment (tspeed90) [reply README.md]: Oh, didn't mean for my readme to be included.

File: src/components/app.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tspeed90) [reply src/components/app.js]: Will take a look a bit later. Thanks!

File: README.md (hunk 1 above)
Comment (haydnba) [reply README.md]: But its definitely better for 'mad-skillzzzz' to be replaced now... it had a good run

**Pull request merged**